code/game/gamemodes/miniantags/revenant/revenant.dm

```diff
@@ -121,9 +121,22 @@
 	remove_from_all_data_huds()
 
 	spawn(5)
-		if(src.mind)
-			src.mind.wipe_memory()
-			src << 'sound/effects/ghost.ogg'
+		if(mind)
+			giveObjectivesandGoals()
+		if(!giveSpells())
+			message_admins("Revenant was created but has no mind. Trying again in ten seconds.")
+			addtimer(CALLBACK(src, .proc/giveSpellsObjectivesordelete), 100)
```

Comment:
tiny thing, but may as well use `10 SECONDS` instead of `100`... the define exists for a reason / readability.

Reply:
Good catch